### PR TITLE
NFC: Switch back to use cats-mtl for casper 

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
Provide a brief description of what this PR does, and why it's needed.

Title says it all. As suggested by @rabbitonweb , we will switch to scalaz together.

### Does this PR relate to an RChain JIRA issue? 
If applicable, add link to corresponding JIRA issue.

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/CliqueOracleTest.scala [hunk 1]
```diff
@@ -64,12 +73,23 @@ class CliqueOracleTest extends FlatSpec with Matchers with BlockGenerator {
...
+        isGenesisSafe              <- TuranOracle[F].isSafe(genesis, 1)
+        _                          = assert(isGenesisSafe)
+        isB2Safe                   <- TuranOracle[F].isSafe(b2, 1)
+        _                          = assert(isB2Safe)
+        isB3Safe                   <- TuranOracle[F].isSafe(b3, 1)
+        _                          = assert(!isB3Safe)
+        isB4SafeConservatively     <- TuranOracle[F].isSafe(b4, 1)
+        _                          = assert(!isB4SafeConservatively)
+        isB4SafeLessConservatively <- TuranOracle[F].isSafe(b4, 0.2f) // Clique oracle would be safe
+        _                          = assert(!isB4SafeLessConservatively)
```
Comment (KentShikama) [inline casper/src/test/scala/coop/rchain/casper/CliqueOracleTest.scala]: @goral09 Your previous PR's suggestion worked well here 😀

File: casper/src/main/scala/coop/rchain/casper/TuranOracle.scala [hunk 2]
```diff
@@ -28,10 +29,18 @@ import scala.collection
  * If all validators are indeed part of the clique, minMaxCliqueWeight will hopefully be equal to total_weight
  * and is_safe will reduce to total_weight >= total_weight and evaluate to true.
  */
-class TuranOracle(blocks: collection.Map[ByteString, BlockMessage],
-                  latestBlocks: collection.Map[ByteString, BlockMessage],
-                  faultToleranceThreshold: Float) {
-  def isSafe(estimate: BlockMessage): Boolean = {
+trait TuranOracle[F[_]] {
```
Comment (birchmd) [inline casper/src/main/scala/coop/rchain/casper/TuranOracle.scala]: This trait seems more general than just for Turan Oracle, same interface would work for any safety oracle. Maybe consider renaming to `SafetyOracle`

Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/TuranOracle.scala]: Done

File: casper/src/test/scala/coop/rchain/casper/CliqueOracleTest.scala (hunk 1 above)
Comment (goral09) [reply casper/src/test/scala/coop/rchain/casper/CliqueOracleTest.scala]: :+1:  :)

File: project/Dependencies.scala [hunk 3]
```diff
@@ -10,6 +10,7 @@ object Dependencies {
   val bitcoinjCore        = "org.bitcoinj"                % "bitcoinj-core"             % "0.14.6"
   val bouncyCastle        = "org.bouncycastle"            % "bcprov-jdk15on"            % "1.58"
   val catsCore            = "org.typelevel"              %% "cats-core"                 % "1.0.1"
+  val catsMtl             = "org.typelevel" %% "cats-mtl-core" % "0.2.1"
```
Comment (rabbitonweb) [inline project/Dependencies.scala]: pls format this

**Pull request merged**